tests/aws/services/events/test_events_targets.py

```diff
@@ -37,6 +37,116 @@
+                    "Action": ["logs:CreateLogStream", "logs:PutLogEvents"],
+                    "Resource": f"{log_group_arn}:*",
+                }
+            ],
+        }
+        policy_name = f"EventBridgePolicy-{short_uid()}"
+        aws_client.logs.put_resource_policy(
+            policyName=policy_name, policyDocument=json.dumps(resource_policy)
+        )
+
+        # Wait for the resource policy to take effect (only needed in AWS)
+        if is_aws_cloud():
+            time.sleep(10)
+
+        # Step 4: Create a rule on the event bus
+        rule_name = f"test-rule-{short_uid()}"
+        rule_response = events_put_rule(
+            Name=rule_name,
+            EventBusName=event_bus_name,
+            EventPattern=json.dumps(TEST_EVENT_PATTERN),
+        )
+        snapshot.match("rule_response", rule_response)
+
+        # Step 5: Add the CloudWatch Logs log group as a target
+        target_id = f"target-{short_uid()}"
+        put_targets_response = aws_client.events.put_targets(
+            Rule=rule_name,
+            EventBusName=event_bus_name,
+            Targets=[
+                {
+                    "Id": target_id,
+                    "Arn": log_group_arn,
+                }
+            ],
+        )
+        snapshot.match("put_targets_response", put_targets_response)
+        assert put_targets_response["FailedEntryCount"] == 0
+
+        # Step 6: Send an event to the event bus
+        event_entry = {
+            "EventBusName": event_bus_name,
+            "Source": TEST_EVENT_PATTERN["source"][0],
+            "DetailType": TEST_EVENT_PATTERN["detail-type"][0],
+            "Detail": json.dumps(EVENT_DETAIL),
+        }
+        put_events_response = aws_client.events.put_events(Entries=[event_entry])
+        snapshot.match("put_events_response", put_events_response)
+        assert put_events_response["FailedEntryCount"] == 0
+
+        # Step 7: Verify that the event was logged to the log group
+        def get_log_events():
+            response = aws_client.logs.describe_log_streams(logGroupName=log_group_name)
+            log_streams = response.get("logStreams", [])
+            if not log_streams:
+                raise Exception("No log streams found")
+            log_stream_name = log_streams[0]["logStreamName"]
+            events_response = aws_client.logs.get_log_events(
+                logGroupName=log_group_name,
+                logStreamName=log_stream_name,
+            )
+            events = events_response.get("events", [])
+            if not events:
+                raise Exception("No log events found")
+            return events
+
+        events = retry(get_log_events, retries=5, sleep=5)
+        snapshot.match("log_events", events)
+
+        # Cleanup: delete the log group
+        aws_client.logs.delete_log_group(logGroupName=log_group_name)
+        aws_client.logs.delete_resource_policy(policyName=policy_name)
```

Comment:
Perhaps rather use the `cleanup` fixture since these resources could end up hanging around if they're created and the test exits before this is reached.